client/app/components/Settings.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { Platform, StyleSheet } from 'react-native';
+import { connect } from 'react-redux';
+import {
+  Item,
+  Label,
+  Input,
+  Switch,
+  Form,
+} from 'native-base';
+
+import Colors from '../constants/Colors';
+import { settingsStrings } from '../constants/Strings';
+import { PTSansText } from '../components/StyledText';
+import { modifyProfile } from '../actions/userActions';
+
+const Settings = ({ userProfile, formOptions, modifyProfile, onProfileModified }) => {
+
+  // TODO: Add check for empty user profile fields
+  const formItems = formOptions.map(option => (
+    <Item key={option.item} floatingLabel>
+      <Label
+        style={{
+          color: !userProfile.isAuthenticated ? 'white' : null,
```

Comment:
I think it would be more readable to make a style object for any component that has conditional styling here. Doing it in place is a bit busy imo.

Reply:
Addressed